go/storage/mkvs/checkpoint/file.go

```diff
@@ -162,22 +162,43 @@ func (fc *fileCreator) GetCheckpoint(ctx context.Context, request *GetCheckpoint
 	return &cp, nil
 }
 
-func (fc *fileCreator) DeleteCheckpoint(ctx context.Context, request *DeleteCheckpointRequest) error {
+func (fc *fileCreator) DeleteCheckpoint(ctx context.Context, version uint16, root node.Root) error {
 	// Currently we only support a single version.
-	if request.Version != checkpointVersion {
+	if version != checkpointVersion {
 		return ErrCheckpointNotFound
 	}
 
-	checkpointDir := filepath.Join(
-		fc.dataDir,
-		strconv.FormatUint(request.Root.Version, 10),
-		request.Root.Hash.String(),
-	)
-	if _, err := os.Stat(checkpointDir); err != nil {
+	versionDir := filepath.Join(fc.dataDir, strconv.FormatUint(root.Version, 10))
+	checkpointDir := filepath.Join(versionDir, root.Hash.String())
+	checkpointFilename := filepath.Join(checkpointDir, checkpointMetadataFile)
+	if err := os.Remove(checkpointFilename); err != nil {
```

Comment:
If you're going to `RemoveAll` the directory that contains the checkpoint file, isn't explicitly removing the checkpoint dir redundant?

Reply:
The idea is that `RemoveAll` could take some time if there are a lot of chunks. If we remove the metadata file first, this will prevent the checkpoint from being listed in `GetCheckpoints` etc. while it is being removed. The alternative would be to rename the checkpoint directory before removing it (and then skipping such directories in `GetCheckpoints`).

In practice it may not make a big difference if removals are fast.

Reply:
Ok, that seems fine.